models/evc.py

```diff
@@ -1058,32 +1079,107 @@ def run_sdntrace(uni):
             return []
         return response.json().get('result', [])
 
-    def check_traces(self):
-        """Check if current_path is deployed comparing with SDN traces."""
-        trace_a = self.run_sdntrace(self.uni_a)
-        if len(trace_a) != len(self.current_path) + 1:
-            log.warning(f"Invalid trace from uni_a: {trace_a}")
-            return False
-        trace_z = self.run_sdntrace(self.uni_z)
-        if len(trace_z) != len(self.current_path) + 1:
-            log.warning(f"Invalid trace from uni_z: {trace_z}")
-            return False
-
-        for link, trace1, trace2 in zip(self.current_path,
-                                        trace_a[1:],
-                                        trace_z[:0:-1]):
-            if compare_endpoint_trace(
-               link.endpoint_a,
-               glom(link.metadata, 's_vlan.value'), trace2) is False:
-                log.warning(f"Invalid trace from uni_a: {trace_a}")
-                return False
-            if compare_endpoint_trace(
-               link.endpoint_b,
-               glom(link.metadata, 's_vlan.value'), trace1) is False:
-                log.warning(f"Invalid trace from uni_z: {trace_z}")
-                return False
+    @staticmethod
+    def run_bulk_sdntraces(uni_list):
+        """Run SDN traces on control plane starting from EVC UNIs."""
+        endpoint = f"{settings.SDN_TRACE_CP_URL}/traces"
+        data = []
+        for uni in uni_list:
+            data_uni = {
+                "trace": {
+                            "switch": {
+                                "dpid": uni.interface.switch.dpid,
+                                "in_port": uni.interface.port_number,
+                            }
+                        }
+                }
+            if uni.user_tag:
+                data_uni["trace"]["eth"] = {
+                                            "dl_type": 0x8100,
+                                            "dl_vlan": uni.user_tag.value,
+                                            }
+            data.append(data_uni)
+        response = requests.put(endpoint, json=data, timeout=30)
+        if response.status_code >= 400:
+            log.error(f"Failed to run sdntrace-cp: {response.text}")
+            return []
+        return response.json()
 
-        return True
+    @staticmethod
+    # pylint: disable=too-many-locals
+    def check_list_traces(list_circuits):
```

Comment:
@gretelliz there's underlying issue that can end up incorrectly activating an EVC, I was using a ring topology, shut down uplinks, the EVC had no paths available and yet, it ended up activating it:

```
2022-12-19 10:23:41,767 - WARNING [kytos.napps.kytos/mef_eline] (thread_pool_app_0) EVC(d07324e141df4c, epl) was not deployed. No available path was found.
kytos $> 2022-12-19 10:24:20,747 - INFO [kytos.napps.kytos/mef_eline] (mef_eline) Starting consistency routine
2022-12-19 10:24:20,802 - INFO [kytos.napps.kytos/mef_eline] (mef_eline) EVC(d07324e141df4c, epl) enabled but inactive - activating
2022-12-19 10:24:20,816 - INFO [kytos.napps.kytos/mef_eline] (mef_eline) Finished consistency routine
```

```
{
    "active": true,
    "archived": false,
    "backup_links": [],
    "backup_path": [],
    "bandwidth": 0,
    "circuit_scheduler": [],
    "creation_time": "2022-12-19T13:10:29",
    "current_path": [],
    "dynamic_backup_path": true,
    "enabled": true,
    "end_date": null,
    "execution_rounds": 0,
    "failover_path": [],
    "flow_removed_at": "2022-12-19T13:23:41",
    "id": "d07324e141df4c",
    "metadata": {},
    "name": "epl",
    "owner": null,
    "primary_constraints": {
        "spf_attribute": "hop"
    },
    "primary_links": [],
    "primary_path": [],
    "queue_id": 7,
    "request_time": "2022-12-19T13:10:29",
    "sb_priority": 5000,
    "secondary_constraints": {},
    "service_level": 7,
    "start_date": "2022-12-19T13:10:29",
    "uni_a": {
        "interface_id": "00:00:00:00:00:00:00:01:1"
    },
    "uni_z": {
        "interface_id": "00:00:00:00:00:00:00:03:1"
    },
    "updated_at": "2022-12-19T13:24:20"
}
```

Reply:
@viniarck  After the last update this problem seems to be solved, I no longer get the error for the mentioned configuration